ui/src/core_components/embed/CorePDF.vue

```diff
@@ -33,26 +60,251 @@ export default {
 				type: FieldType.Text,
 				desc: "A valid URL. Alternatively, you can provide a state reference to a packed PDF file.",
 			},
+			highlights: {
+				name: "Highlights",
+				default: JSON.stringify([]),
+				desc: "A list of highlights to be applied to the PDF as a JSON array of strings.",
+				type: FieldType.Object,
+			},
+			selectedMatch: {
+				name: "Selected highlight match",
+				default: null,
+				desc: "The index of the selected highlight match.",
+				type: FieldType.Number,
+			},
+			page: {
+				name: "Page",
+				type: FieldType.Number,
+				desc: "The page to be displayed.",
+			},
 			cssClasses,
```

Comment:
Please also add separatorColor, backgroundColor and primaryTextColor so that users can configure these.